test/e2e/lib/components/abstract-editor-component.js

```diff
@@ -0,0 +1,47 @@
+import { By } from 'selenium-webdriver';
+import AsyncBaseContainer from '../async-base-container';
+import * as driverHelper from '../driver-helper';
+
+/**
+ * Abstract class containing shared code for editor-related classes.
+ */
+export default class AbstractEditorComponent extends AsyncBaseContainer {
+	/**
+	 * Insert a child block into a parent block using the inline block inserter.
+	 *
+	 * @param {string} name the name of the block to insert as a child.
+	 * @param {string} container the name of the container block to insert the child block into.
+	 */
+	async insertBlockOrPatternViaBlockAppender( name, container = 'Group' ) {
+		const containerBlockId = await this.addBlock( container );
+		await this.runInCanvas( async () => {
+			const blockAppenderLocator = By.css(
+				`#${ containerBlockId } .block-editor-button-block-appender`
+			);
+			await driverHelper.clickWhenClickable( this.driver, blockAppenderLocator );
+		} );
+
+		const quickInserterSearchInputLocator = By.css(
+			`.block-editor-inserter__quick-inserter .components-search-control__input`
+		);
+
+		const patternItemLocator = By.css(
+			'.block-editor-inserter__quick-inserter .block-editor-block-types-list__item, .block-editor-inserter__quick-inserter .block-editor-block-patterns-list__item'
+		);
+
+		await driverHelper.setWhenSettable( this.driver, quickInserterSearchInputLocator, name );
+		await driverHelper.clickWhenClickable( this.driver, patternItemLocator );
+	}
+
+	/**
+	 * Set the running context for operating on the DOM, i.e switch to the right
+	 * iframe. By default, it's defined here as a noop, but can be overriden in
+	 * subclasses with a proper implementation, if needed.
+	 *
+	 * @param {Function} cb  The callback setting the right document context.
+	 * @returns {*} depends on the return value of the cb function.
+	 */
+	async runInCanvas( cb ) {
+		return await cb();
+	}
```

Comment:
I'm wondering why this one's used in the `site-editor-component.js` but not in the `gutenberg-editor-component.js`. Is the site-editor extra iframed? Anyway, if it turns out to be only the case for the site-editor, then we shouldn't probably be abstracting it. What do you think? cc @david-szabo97

Reply:
To add to that, I think we could further improve this, and put some additional thought on the `runInCanvas` function. I think it should be named `runInContext` or something, but I'd rather leave it for another PR :). Also, if you have any other ideas on how better solve this, let me know!